### ISSUE TITLE
Feature request: add a `local_init_feed_dict` to `tf.train.Scaffold`

### ISSUE THREAD

**Issue by lsorber** [tensorflow/tensorflow#11665]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.2.0
- **Python version**: 3.6.1
- **Bazel version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: See below.

### Describe the problem
Feature request: add a `local_init_feed_dict` to `tf.train.Scaffold`. It would be useful to be able to create local variables (which are not saved or restored) and have them initialized by a `tf.train.MonitoredTrainingSession` with a `feed_dict`. In the example below, the variable `X_var` is forced to be part of the `GLOBAL_VARIABLES` collection in order to be able to initialize the variable with a `feed_dict`. This has the undesirable consequence that the variable will be saved to disk.

### Source code / logs
```python
import tensorflow as tf
import numpy as np

# Data that we wish to sample, but not save to disk.
X = np.eye(15, dtype=np.float32)

# Create a graph that samples rows from X randomly.
graph = tf.Graph()
with graph.as_default():
    X_init = tf.placeholder(tf.float32, shape=X.shape)
    # Here, we want to use tf.GraphKeys.LOCAL_VARIABLES,
    # but can't because there is no feed_dict for that collection in tf.train.Scaffold.
    X_var = tf.Variable(X_init, trainable=False, collections=[tf.GraphKeys.GLOBAL_VARIABLES])
    queue = tf.RandomShuffleQueue(
        capacity=X.shape[0],
        min_after_dequeue=1,
        dtypes=[tf.float32],
        shapes=[X.shape[1]])
    enqueue_op = queue.enqueue_many([X_var])
    row = queue.dequeue()

# Sample a few rows from X.
with graph.as_default():
    sess_params = {
        'scaffold': tf.train.Scaffold(
            init_feed_dict={X_init: X},
            init_fn=lambda scaffold, sess: sess.run(enqueue_op))
    }
    with tf.train.MonitoredTrainingSession(**sess_params) as sess:
        print(sess.run(row))
        print(sess.run(row))
```

**Comment by reedwm** [comment]
Is there any reason a `local_init_feed_dict` would be preferable to directly initializing `X_var` with `np.eye(...)`, without using a `tf.placeholder`? In that case, you could initialize `X_var` by passing `local_init_op=tf.initialize_local_variables()` to `Scaffold` I'm not sure a feed dict for local variables would be helpful.

**Comment by lsorber** [comment]
@reedwm Wouldn't that approach limit the size of `X` to 2 GB though? With the placeholder approach, the data in `X` is not part of the graph.

**Comment by reedwm** [comment]
Good point, you do need to use a feed dict to initialize variables larger than 2 GB.

/CC @ispirmustafa, what do you think?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by ispirmustafa** [comment]
Hi @reedwm, @lsorber 
You can use `Scaffold.init_fn` to initialize a variable based on feed_dict.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activityand the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
A member of the TensorFlow organization has replied after the stat:awaiting tensorflower label was applied.

**Comment by reedwm** [comment]
Closing since @ispirmustafa's suggestion resolves the problem

**Comment by lhlmgr** [comment]
@ispirmustafa, @reedwm  sorry for commenting on a closed issue.
But I found maybe a use-case where its not possible to use `init_fn`.
Consider the following programm:

```
x_train, y_train = zip(*D_train)

features_placeholder = tf.placeholder(x_train.dtype, x_train.shape)
labels_placeholder = tf.placeholder(y_train.dtype, y_train.shape)
dataset = tf.data.Dataset.from_tensor_slices((features_placeholder, labels_placeholder))
dataset = dataset.batch(64)
iterator = dataset.make_initializable_iterator()

scaffold_train = tf.train.Scaffold(local_init_op=tf.group(tf.local_variables_initializer(),
                                                          tf.tables_initializer(),
                                                          iterator.initializer))
```

When training the model from scratch it would be possible to use `init_fn`, however, when restoring the model from checkpoint, I need the `local_init_op` to initialize the `iterator` with the placeholders and therefore a `feed_dict` for the `local_init_op`.

Of course, one could easily create TFRecord-files and use those, but for quickly testing on numpy arrays this would be very helpful.
